src/features/device-state/state-patch-utils.ts

```diff
@@ -58,29 +61,32 @@ const truncateText = (
 		.trim();
 };
 
-type ValidPatchFields = Array<
-	(typeof v3ValidPatchFields)[number] | (typeof v2ValidPatchFields)[number]
->;
+type ValidPatchField =
+	| (typeof v3ValidPatchFields)[number]
+	| (typeof v2ValidPatchFields)[number];
 
-const defaultShortTextFieldsToTruncate: ValidPatchFields = [
-	'ip_address',
-	'mac_address',
-];
-export const truncateShortTextFields = (
-	object: Dictionary<any>,
-	keysToTruncate: ValidPatchFields = defaultShortTextFieldsToTruncate,
-) => {
-	for (const key of keysToTruncate) {
-		if (
-			typeof object[key] !== 'string' ||
-			object[key].length <= SHORT_TEXT_LENGTH
-		) {
+const constrainedDeviceTextFields = [
+	['ip_address', 2000, ADDRESS_DELIMITER],
+	['mac_address', 900, ADDRESS_DELIMITER],
```

Comment:
We are going to add more stuff in this list in https://github.com/balena-io/open-balena-api/pull/2026 but since that's going to be a huge PR anyway, I decided to prepare this helper already in this PR which is smaller.